apps/funding-platform/src/request/request.service.ts

```diff
@@ -0,0 +1,141 @@
+import { DBInstance } from "../db";
+import {
+  getRequestsByAccessToken as getRequestsByAccessTokenDB,
+  saveRequest as saveRequestDB,
+  getRequest as getRequestDB,
+  updateRequest as updateRequestDB,
+  deleteRequest as deleteRequestDB,
+  getRequests as getRequestsDB,
+} from "../db";
+import { CreateRequest, QueryRequest, UpdateRequest } from "./dto";
+import { utils } from "rpch-common";
+
+const { log, logError } = utils.createLogger([
+  "funding-platform",
+  "request-service",
+]);
+
+/**
+ * An abstraction layer for requests to interact with db.
+ * @param db holds all methods to interact with db
+ */
+export class RequestService {
+  constructor(private db: DBInstance) {}
+
+  public async createRequest(params: {
+    address: string;
+    amount: string;
+    chainId: number;
+    accessTokenHash: string;
+  }) {
+    try {
+      log("Creating request...");
+      const now = new Date(Date.now());
+      const createRequest: CreateRequest = {
+        amount: params.amount,
+        accessTokenHash: params.accessTokenHash,
+        nodeAddress: params.address,
+        chainId: params.chainId,
+        createdAt: now.toISOString(),
+        requestId: Math.floor(Math.random() * 1e6),
```

Comment:
I would use something more random here, but lets fix this as a patch after we discuss https://github.com/Rpc-h/RPCh/issues/82